hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java

```diff
@@ -1976,4 +1976,19 @@
 	 * @since 5.4
 	 */
 	String SEQUENCE_INCREMENT_SIZE_MISMATCH_STRATEGY = "hibernate.id.sequence.increment_size_mismatch_strategy";
+
+	/**
+	 * <p>
+	 * When you use {@link javax.persistence.InheritanceType#JOINED} strategy for inheritance mapping and query
+	 * a value from an entity, all superclass tables are joined in the query regardless you need them. With
+	 * this setting set to true only superclass tables which are really needed are joined.
+	 * </p>
+	 * <p>
+	 * The default value is true.
+	 * </p>
+	 *
+	 * @Since 5.4
+	 */
+	String OMIT_JOIN_OF_SUPERCLASS_TABLES = "hibernate.query.omit_join_of_superclass_tables";
```

Comment:
I'm not a native speaker but the option name sounds weird to my ear. Any native speaker with a better proposal?

I would have gone with something like `lenient_superclass_table_join` because you don't really always omit them, you are just considering only the useful ones.

@sebersole you're usually good with finding names. Any better proposal?

Reply:
I don't really like the property name either. Your suggestion looks fine.